src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs

```diff
@@ -133,5 +133,34 @@ internal bool HasClientCredentials
         /// See https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal.
         /// </summary>
         public string? UserAssignedManagedIdentityClientId { get; set; }
+
+        /// <summary>
+        /// Applies to a web API: does the web API allow app-only tokens?.
+        /// </summary>
+        /// <remarks>
+        /// In the app registration you'll need to opt-in for the idtyp claim.
+        /// </remarks>
+        public bool ApiAllowsAppOnlyTokens { get; set; } = true;
+
+        /// <summary>
+        /// Applies to a web API: does the web API allow guest accounts?.
+        /// </summary>
+        /// <remarks>
+        /// In the app registration for your web API, you'll need to opt-in for the acct claim.
+        /// </remarks>
+        public bool ApiAllowsGuestAccounts { get; set; } = true;
```

Comment:
guests are identified with a WID in the token. There should be no need to handle them explicitly, Authorize on the WID, or lack of it.

Reply:
WIDS claim is unreliable.  We want to do as much as we can using only claims in the token, and sometimes you ne4ed to call Graph for wids and groups.